dipy/reconst/tests/test_dki_micro.py

```diff
@@ -0,0 +1,241 @@
+    RDEc = RDE.copy()
+    DWIsimc = DWIsim.copy()
+
+    FIEc[0, 0, 0] = 0
+    RDIc[0, 0, 0] = 0
+    ADIc[0, 0, 0] = 0
+    ADEc[0, 0, 0] = 0
+    Torc[0, 0, 0] = 0
+    RDEc[0, 0, 0] = 0
+    DWIsimc[0, 0, 0, :] = 0
+    mask = np.ones((2., 2., 2.))
+    mask[0, 0, 0] = 0
+
+    dkiF = dkiM.fit(DWIsimc)
+    awf = dki_micro.axonal_water_fraction(dkiF.model_params, sphere,
+                                          gtol=1e-5)
+    assert_almost_equal(awf, FIEc)
+
+    # Extra-cellular and intra-cellular components
+    edt, idt = dki_micro.diffusion_components(dkiF.model_params, sphere,
+                                              awf=awf)
+    EDT = eig_from_lo_tri(edt)
+    IDT = eig_from_lo_tri(idt)
+    assert_array_almost_equal(EDT[..., 0], ADEc)
+    assert_array_almost_equal(EDT[..., 1], RDEc)
+    assert_array_almost_equal(EDT[..., 2], RDEc)
+    assert_array_almost_equal(IDT[..., 0], ADIc)
+    assert_array_almost_equal(IDT[..., 1], RDIc)
+    assert_array_almost_equal(IDT[..., 2], RDIc)
+
+    # Check when mask is given
+    dkiF = dkiM.fit(DWIsim)
+    awf = dki_micro.axonal_water_fraction(dkiF.model_params, sphere,
+                                          gtol=1e-5, mask=mask)
+    assert_almost_equal(awf, FIEc)
+
+    # Extra-cellular and intra-cellular components
+    edt, idt = dki_micro.diffusion_components(dkiF.model_params, sphere,
+                                              awf=awf, mask=mask)
+    EDT = eig_from_lo_tri(edt)
+    IDT = eig_from_lo_tri(idt)
+    assert_array_almost_equal(EDT[..., 0], ADEc)
+    assert_array_almost_equal(EDT[..., 1], RDEc)
+    assert_array_almost_equal(EDT[..., 2], RDEc)
+    assert_array_almost_equal(IDT[..., 0], ADIc)
+    assert_array_almost_equal(IDT[..., 1], RDIc)
+    assert_array_almost_equal(IDT[..., 2], RDIc)
+
+    # Check class object
+    wmtiM = dki_micro.KurtosisMicrostructureModel(gtab_2s, fit_method="WLS")
+    wmtiF = wmtiM.fit(DWIsim, mask=mask)
+    assert_almost_equal(wmtiF.awf, FIEc)
+    assert_almost_equal(wmtiF.axonal_diffusivity, ADIc)
+    assert_almost_equal(wmtiF.hindered_ad, ADEc)
+    assert_almost_equal(wmtiF.hindered_rd, RDEc)
+    assert_almost_equal(wmtiF.tortuosity, Torc)
+
+
+def test_dki_micro_predict():
+    dkiM = dki_micro.KurtosisMicrostructureModel(gtab_2s)
+    dkiF = dkiM.fit(DWIsim)
+
+    # Check predict of KurtosisMicrostruturalModel
+    pred = dkiM.predict(dkiF.model_params)
+    assert_array_almost_equal(pred, DWIsim_all_taylor)
+
+    pred = dkiM.predict(dkiF.model_params, S0=100)
+    assert_array_almost_equal(pred, DWIsim_all_taylor * 100)
+
+    # Check predict of KurtosisMicrostruturalFit
+    pred = dkiF.predict(gtab_2s, S0=100)
+    assert_array_almost_equal(pred, DWIsim_all_taylor * 100)
+
+
+def test_dki_micro_awf_only():
+    dkiM = dki_micro.KurtosisMicrostructureModel(gtab_2s)
+    dkiF = dkiM.fit(DWIsim, awf_only=True)
+    awf = dki_micro.axonal_water_fraction(dkiF.model_params)
+    assert_almost_equal(awf, FIE)
```

Comment:
Hi @arokem! How is the best way to test errors in fit object properties?

I already tried the bellow lines of code, but it didn't work since errors are raised before the function evaluated the property call.

from nose.tools import assert_raises
dkiM = dki_micro.KurtosisMicrostructureModel(gtab_2s)
dkiF = dkiM.fit(DWIsim, awf_only=True)
assert_raises(ValueError, dkiF.hindered_rd)